Doc/library/csv.rst

```diff
@@ -56,8 +56,9 @@ The :mod:`csv` module defines the following functions:
 .. function:: reader(csvfile, dialect='excel', **fmtparams)
 
    Return a reader object which will iterate over lines in the given *csvfile*.
-   *csvfile* can be any object which supports the :term:`iterator` protocol and returns a
-   string each time its :meth:`!__next__` method is called --- :term:`file objects
+   *csvfile* can be any object which supports the :term:`iterable` protocol and returns an
+   iterator that returns a string each time its :term`iterator`'s :meth:`!__next__` method
+   is called when its :meth:`!__iter__` method is called --- :term:`file objects
```

Comment:
I would suggest instead changing the original with the following minimal change:
   `...can be any iterable object whose iterator returns a string each time its...`